pkg/git/github.go

```diff
@@ -0,0 +1,47 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/codefresh-io/cli-v2/pkg/log"
+)
+
+func VerifyGitHubTokenScope(ctx context.Context, token string) (bool, error) {
+	log.G(ctx).Info("Verifing your git token")
+
+	req, _ := http.NewRequest("HEAD", "https://api.github.com/", nil)
+	req.Header.Set("Authorization", fmt.Sprintf("token %s", token))
+
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
you need to use `ctx` when sending the request.

Reply:
please explain

Reply:
https://pkg.go.dev/net/http#NewRequestWithContext